app/src/main/AndroidManifest.xml

```diff
@@ -3,6 +3,7 @@
     package="com.example.lhi06.residue">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.WRITE_INTERNAL_STORAGE"/>
```

Comment:
Make sure you are not copying things straight from stackoverflow without knowing what they do or if they are needed. 
You would only need a permission if you were saying to external storage.